charts/argo/values.yaml

```diff
@@ -322,9 +322,207 @@ artifactRepository:
 
 
 # NOTE: These are setting attributes for the `minio` optional dependency
+# According to Minio, deploying Minio with this dependency chart is no longer supported. Therefor we recommend using the minio-operator chart or provision minio tenants as standalone manifests (raw Chart).
+# More information: https://helm.min.io/
+# This dependency might be removed in the future
 minio:
   # If set to true then chart installs minio and generate according artifactRepository section in workflow controller config map
   install: false
   defaultBucket:
     enabled: true
     name: argo-artifacts
+
+# Minio Operator: https://github.com/minio/operator/tree/master/helm/minio-operator
+# This deploys the minio operator with argo as dependency. This might not be suitable for production environments, since the minio-operator might already be deployed and managed by another helm release. In this case you should provision a new minio tenants with the extraResources key.
+minio-operator:
+  install: false
+
+# Raw Chart: https://artifacthub.io/packages/helm/main/raw
+# Allows to add standalone kubernetes manifests into a helm release.
+raw:
+  ## Enable rendering of raw kubernetes manifests
+  enabled: false
+  # Converted Example Minio Tenant Example
+  # Reference: https://github.com/minio/operator/blob/master/examples/tenant.yaml
+  resources:
+  ## Secret to be used as MinIO Root Credentials
+  - apiVersion: v1
+    kind: Secret
+    metadata:
+      name: minio-creds-secret
+    type: Opaque
+    data:
+      ## Access Key for MinIO Tenant, base64 encoded (echo -n 'minio' | base64)
+      accesskey: bWluaW8=
+      ## Secret Key for MinIO Tenant, base64 encoded (echo -n 'minio123' | base64)
+      secretkey: bWluaW8xMjM=
+  ## Secret to be used for MinIO Console
+  - apiVersion: v1
+    kind: Secret
```

Comment:
What I do not like here is that `resources` is a list. When we need to customize _(and we do need this)_ it from external _(eighter `--set` or via `--values`)_, we need to define the whole block again since lists cannot be merged...

Reply:
That assumption is missing the point of this chart. It's not supposed to be configurable by external values, since this just adds to functionality to add lose resources. If you would want to achieve what you are looking for, you would be using `.templates`. So yes, it's intentional that this block must be defined again. and what's with using `-f lose-resources.yaml`.

Reply:
Using raw is too prescriptive but luckily the upstream minio-operator chart includes a single tenant configuration for those wanting out of the box functionality. We should include a comment `tentnant: []` which disables the operator from generating a tenant for those looking to define the cr's in their `templates/` directory. That should handle really all the use-cases and keep values.yaml lean and to the point.

Reply:
That does not make too much sense. Well ofc you can stage with the operator a tenant. But what if you manage the operator with another helm release, as most production environments do. You are not able to just add a tenant without also deploying the operator with the minio-operator chart.  For that use-case we use the raw chart.

I will changed the values to make it more clear.

Reply:
You just pass in a blank tenant list. Please see the default here: https://github.com/minio/operator/blob/master/helm/minio-operator/values.yaml#L31

Reply:
Hi @tsunamishaun,
Also in my opinion it is absolutely correct what @oliverbaehler tries to highlight:
Deploying the operator as a subchart from an application chart is very uncommon. The operator normally operates in a different namespace and acts as a clusterlevel operator. Also the operator creates cluster-level CRDs.
Therefore we disable the operator deployment by default. But disabling the subchart also results that we cannot deploy MinIO Tenats via this subchart. He solves this via a raw chart which is a viable approach. An alternative approach would be to define serveral resources in the `templates` directory of the argo workflow chart.

Reply:
Correct, Is this implementation as is acceptable for you @mkilchhofer ?

Reply:
@oliverbaehler this is shaping up nicely, good work. @mkilchhofer thank you for emphasizing the direction, I myself wouldn't rely on the argo chart to give me cr's for a minio tenant but perhaps others would? To that point though, having 200+ lines of unused raw in values.yaml is simplified quite drastically by the reference to the upstream examples `tenants.yaml` and may be unneeded. There are also 10+ ways to deploy a tenant so keeping that open seems like the best choice? (https://github.com/minio/operator/tree/master/examples)

Reply:
Well, who would rely on maging the minio operator with the argo chart? I wouldn't. But we should always try to achieve a gerneric approach, so that everyone can work with the chart and doesn't have to implement his own patched version or first publiush changes to make it work.I removed the samples to make the values slimer.